cmd/bom/cmd/root.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/release/pkg/spdx"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/release-utils/log"
+)
+
+var rootCmd = &cobra.Command{
+	Short: "bom: A tool for working with SPDX manifests",
+	Long: `bom (Bill of Materials)
+
+bom is a little utility that lets software authors generate
+SPDX manifests to describe the contents of a release. The
+SPDX manifests provide a way to list and verify all items
+contained in packages, images, and individual files while
+packing the data along with licensing information.
+
+bom is still in its early stages and it is an effort to open
+the libraries developed for the Kubernetes SBOM for other 
+projects to use.
+
+`,
+	Use:          "bom",
+	SilenceUsage: false,
+	// SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+}
+
+var generateCmd = &cobra.Command{
+	Short: "bom generate → Create SPDX manifests",
+	Long: `bom → Create SPDX manifests
+
+generate is the bom subcommand to generate SPDX manifests.
+Currently supports creating SBOM for files, images, and docker
+archives (images in tarballs). Supports remote pulling from
+registries.
+
+`,
+	Use:               "generate",
+	SilenceUsage:      true,
+	SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return generateBOM(genOpts)
+	},
+}
```

Comment:
perhaps this belongs in a separate file? e.g. `cmd/bom/cmd/generate.go`

Reply:
Good catch, Wilson!

Reply:
Yes, I left this in the same file as it is still pretty small. I'll split it now to avoid confusion.

Reply:
Done